recipe/meta.yaml

```diff
@@ -92,6 +93,7 @@ outputs:
         - __cuda >={{ cuda_compiler_version_min }}  # [cuda_compiler_version != "None"]
       run_constrained:
         - apache-arrow-proc =*={{ build_ext }}
+        - numpy
```

Comment:
When adding numpy as is here to `run_constrained`, will it still pick up the version range from numpy's `run_export`? Or if listing numpy here, do we need to manually add the version range?

Reply:
If you ignore the run-export, then nothing will be picked up. If you want to impose a range, we'll have to "manually" add the version constraints from the run-export here.

Reply:
And doing something like `pin_compatible('numpy')` also won't work? (I see pin_compatible being used in some other recipes in the run_constrained section)

Reply:
`pin_compatible` is just a fancy way of saying "take version `X.Y` of this dependency at build time, and create a runtime constraint `>=X.Y,<{X+1}` (and so on if the max_pin is tighter than major-level)

That's actually not fully accurate for numpy, which [uses](https://github.com/conda-forge/numpy-feedstock/blob/1b3ce6a006b96f3356d252ac874907398a88a910/recipe/meta.yaml#L51)
```
numpy >={{ default_abi_level }},<3
```
where -- very unusually -- the runtime lower bound is allowed to be lower than at build time. For numpy 2.1, it's `>=1.21,<3` for example.

However, using `pin_compatible("numpy")` would still be good enough. It'd be tighter than it needs to be (effectively forcing pyarrow users to upgrade to numpy 2.0, whereas before they were still able to use 1.x), but perhaps that's not a big deal.

Reply:
> It'd be tighter than it needs to be (effectively forcing pyarrow users to upgrade to numpy 2.0, whereas before they were still able to use 1.x), but perhaps that's not a big deal.

I think I would prefer not forcing users to upgrade to numpy  >= 2 through an update of pyarrow (we deliberately kept compatibility with 1.x in pyarrow)

Reply:
It should be possible to do something like
```
- {{ pin_compatible("numpy", lower_bound="1.21", max_pin="x") }}
```